internal/tools/restart.go

```diff
@@ -0,0 +1,124 @@
+package tools
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"github.com/tektoncd/mcp-server/internal/params"
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	pipelineclient "github.com/tektoncd/pipeline/pkg/client/injection/client"
+	pipelineruninformer "github.com/tektoncd/pipeline/pkg/client/injection/informers/pipeline/v1/pipelinerun"
+	taskruninformer "github.com/tektoncd/pipeline/pkg/client/injection/informers/pipeline/v1/taskrun"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func restartPipelineRun() server.ServerTool {
+	return server.ServerTool{
+		Tool: mcp.NewTool("restart_pipelinerun",
+			mcp.WithDescription("Restart a PipelineRun"),
+			mcp.WithString("name", mcp.Required(),
+				mcp.Description("Name or Reference of the PipelineRun to restart"),
+			),
+			mcp.WithString("namespace",
+				mcp.Description("Namespace where the PipelineRun is located"),
+				mcp.DefaultString("default"),
+			),
+			// TODO add "parameters" objects
+		),
+		Handler: handlerRestartPipelineRun,
+	}
+}
+
+func handlerRestartPipelineRun(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+	name := request.Params.Arguments["name"].(string)
```

Comment:
I dropped the check on this since we mark it as required in the schema, let me know what you think!

Reply:
That raise a good point. I *think* the check as it was written was a type check (on string) and not a "key present in the map" check. But that means the condition wasn't correct though.